WoodStar/WoodStar/WoodStarConnection.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            var host = Dns.GetHostEntry(_serverIp);
+            var ipAddress = host.AddressList[1];
+            var remoteEP = new IPEndPoint(ipAddress, 1433);
+            var socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+            await socket.ConnectAsync(remoteEP);
+            var networkStream = new NetworkStream(socket);
+            _tdsChannel = new TdsChannel(networkStream);
+            await _tdsChannel.Prelogin();
+
+            var interceptingStream = new InterceptingStream(networkStream);
+            var sslStream = new SslStream(interceptingStream, leaveInnerStreamOpen: false);
+            await sslStream.AuthenticateAsClientAsync(new SslClientAuthenticationOptions()
+            {
+                TargetHost = _serverIp,
+                RemoteCertificateValidationCallback = (_, _, _, _) => true,
+            });
+            interceptingStream.StopIntercepting();
+            _tdsChannel.SetSslStream(sslStream);
+            await _tdsChannel.Login(_username, _password, _database);
+        }
+
+        public Task ExecuteSqlAsync(string sql)
+        {
+            return _tdsChannel.ExecuteSqlAsync(sql);
+        }
+
+        private sealed class InterceptingStream : Stream
+        {
+            private readonly Stream _stream;
+
+            private bool _intercept;
+            private int _packetBytes;
+
+            public InterceptingStream(Stream stream)
+            {
+                _stream = stream;
+                _intercept = true;
+            }
+
+            public void StopIntercepting() => _intercept = false;
+
+            public override bool CanRead => _stream.CanRead;
+
+            public override bool CanSeek => _stream.CanSeek;
+
+            public override bool CanWrite => _stream.CanWrite;
+
+            public override long Length => _stream.Length;
+
+            public override long Position { get => _stream.Position; set => _stream.Position = value; }
+
+            public override void Flush() => _stream.Flush();
+
+            public override Task FlushAsync(CancellationToken cancellationToken) => _stream.FlushAsync(cancellationToken);
+
+            public override int Read(byte[] buffer, int offset, int count) => _stream.Read(buffer, offset, count);
+
+            public override long Seek(long offset, SeekOrigin origin) => _stream.Seek(offset, origin);
+
+            public override void SetLength(long value) => _stream.SetLength(value);
+
+            public override void Write(byte[] buffer, int offset, int count) => _stream.Write(buffer, offset, count);
+
+            public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
+            {
+                if (_intercept)
+                {
+                    if (_packetBytes == 0)
+                    {
+                        // Process header
+                        var header = new byte[TdsHeader.HeaderSize];
```

Comment:
It's possible to remove the array allocation and just use the buffer. It's going to be overwritten anyway...